server/services/store/mattermostauthlayer/mattermostauthlayer.go

```diff
@@ -868,7 +886,12 @@ func (s *MattermostAuthLayer) GetMembersForBoard(boardID string) ([]*model.Board
 		existingMembers[m.UserID] = true
 	}
 	for _, m := range implicitMembers {
-		if !existingMembers[m.UserID] {
+		// No synthetic memberships for guests
+		user, err := s.GetUserByID(m.UserID)
```

Comment:
This is going to be really awful in performance, this is generating one DB roadtrip for each implicit memberships here.

Reply:
My suggestion here is to use the original query to extract the data, another extra join to get the User information, now know if the user there is a guest or not.